packages/nimble-components/src/rich-text/models/mention-extension-configuration.ts

```diff
@@ -1,18 +1,24 @@
+import { ViewTemplate, html } from '@microsoft/fast-element';
 import type { MentionInternals } from '../../rich-text-mention/base/models/mention-internals';
-import type { MentionUpdateEmitter } from '../../rich-text-mention/base/types';
 import { mentionPluginPrefix } from '../editor/types';
+import type {
+    MappingConfigs,
+    MentionUpdateEmitter
+} from '../../rich-text-mention/base/types';
 
 /**
  * A configuration object for a Mention extension, to be used by the editor for loading mention plugins in tiptap.
  * This object maintains the necessary internal values for loading mention extension.
  */
 export class MentionExtensionConfiguration {
     private static instance = 0;
-
+    public readonly iconTemplate: ViewTemplate;
     public readonly viewElement: string;
     public readonly character: string;
     public readonly name: string;
     public readonly key: string;
+    public readonly label: string;
```

Comment:
We should propagate the anem buttonLabel from the other discussions. Also buttonLabel will be the first observable property that is part of MentionExtensionConfiguration so it needs `isObservedMentionInternalsProperty` for the buttonLabel and follow the handleChange patter from the editor integration PR.

Reply:
Updated